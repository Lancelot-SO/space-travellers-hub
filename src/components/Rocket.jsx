import React from 'react';

export default function Rocket() {
  return (
    <div>
      <h1>rocket</h1>
    </div>
  );
}
