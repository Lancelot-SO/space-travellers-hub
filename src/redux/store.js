import { configureStore, combineReducers } from '@reduxjs/toolkit';
import { createLogger } from 'redux-logger';
import rocketReducer from './rocket/rocketSlice';
import missionsReducer from './missions/missionsSlice';

const logger = createLogger({
  predicate: (getState, action) => action.type !== 'SOME_ACTION',
  collapsed: true,
});

const store = configureStore({
  reducer: combineReducers({
    rockets: rocketReducer,
    missions: missionsReducer,
  }),
  middleware: (getDefaultMiddleware) => getDefaultMiddleware().concat(logger),
});

export default store;
