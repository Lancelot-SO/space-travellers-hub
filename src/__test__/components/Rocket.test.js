import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import Rocket from '../../components/Rocket';
import store from '../../redux/store';

describe('MissionsTable', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Rocket />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
