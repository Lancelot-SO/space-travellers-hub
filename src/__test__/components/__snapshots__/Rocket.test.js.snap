// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MissionsTable renders correctly 1`] = `
<div
  className="rocketflex"
>
  <div
    className="imgdiv"
  >
    <img
      className="img"
    />
  </div>
  <div>
    <h1
      className="name"
    />
    <p
      className="desc"
    />
    <button
      className="reservebtn"
      onClick={[Function]}
      type="button"
    >
      Reserve Rocket
    </button>
  </div>
</div>
`;
